### PR TITLE
Adds TensileCreateLibrary cli reference docs

### PR DESCRIPTION
**Pull request by ellosel**
Adds the command line interface reference documentation for TensileCreateLibrary intended for v0 documentation.

### CODE REVIEW

File: docs/src/cli-reference/TensileCreateLibrary.rst [hunk 1]
```diff
@@ -84,6 +86,20 @@ Options
...
+
+\<LOGIC DIRECTORY\>
+    Absolute path for logic files. These files are generally found in one of two ways: (i) they are
+    generated via the `Tensile` program and placed in the build directory under *3_LibraryLogic* (see :ref:`quick-start-example`).
+    (ii) they are found within a project that hosts pre-generated logic files, e.g., `rocBLAS <https://github.com/ROCm/rocBLAS/tree/develop/library/src/blas3/Tensile/Logic>`_.
+\<OUTPUT DIRECTORY\>
+    Absolute or relative path to the output directory where build artifacts are placed.
+\<RUNTIME LANGUAGE\>
+    One of: {OCL, HIP, HSA}
+
```
Comment (ellosel) [inline docs/src/cli-reference/TensileCreateLibrary.rst:102]: I don't know if these qualify as options. We should maybe have a separate section for these.

File: docs/src/cli-reference/TensileCreateLibrary.rst [hunk 2]
```diff
@@ -84,6 +86,20 @@ Options
     Output master solution index in csv format including number 
     of kernels per architecture post build in csv format.
 
+Arguments
```
Comment (ellosel) [inline docs/src/cli-reference/TensileCreateLibrary.rst:89]: Can we move these above the options? I feel like the required arguments should come first.

File: Tensile/TensileCreateLibrary.py [hunk 3]
```diff
@@ -1071,7 +1071,7 @@ def splitExtraParameters(par):
   argParser.add_argument("LogicPath",       help="Path to LibraryLogic.yaml files.")
   argParser.add_argument("OutputPath",      help="Where to write library files?")
   argParser.add_argument("RuntimeLanguage", help="Which runtime language?", choices=["OCL", "HIP", "HSA"])
-  argParser.add_argument("--cxx-compiler",           dest="CxxCompiler",       choices=["hipcc"],       action="store", default="hipcc")
+  argParser.add_argument("--cxx-compiler",           dest="CxxCompiler",       choices=["amdclang++, hipcc"],       action="store", default="hipcc")
```
Comment (ellosel) [inline Tensile/TensileCreateLibrary.py:1074]: This isn't an option yet... we should remove here and in the docs I suppose.

File: docs/sphinx/_toc.yml.in [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
-      - file: src/concepts/kernels
-      - file: src/concepts/library-creation
+        - file: src/concepts/benchmarking
+        - file: src/concepts/tuning
+        - file: src/concepts/kernels
+        - file: src/concepts/library-creation
     - file: src/api-reference
     - file: src/cli-reference
+      entries: 
+      - file: src/cli-reference/TensileCreateLibrary
```
Comment (bstefanuk) [inline docs/sphinx/_toc.yml.in:16]:
```suggestion
        - file: src/cli-reference/TensileCreateLibrary
```
Match file indentation under entries above.

File: docs/src/cli-reference.rst [hunk 5]
```diff
@@ -4,8 +4,29 @@
...
+   .. table:: Command-line syntax
+      :width: 70%
+      :align: center
+
+      ================================= ==================================
+      Notation                          Description
+      ================================= ==================================
+      ``<Text inside angle brackets>``  Required argument.
+      ``[Text inside square brackets]`` Optional arguments.
+      ``{Test inside braces}``          Set of arguments. One is required.
```
Comment (bstefanuk) [inline docs/src/cli-reference.rst:30]: Are the curly braces intended to include multiple optional arguments like `[OPTION ...]`? If not, we should add this row for clarity.

File: docs/src/cli-reference/TensileCreateLibrary.rst [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+invoking *TensileCreateLibrary*. In some cases, such as ``--seprate-architectures``
+no arguments are required; whereas, for ``--jobs`` an argument is required.
+
+.. code-block::
+
+    TensileCreateLibrary --separate-architectures --jobs=32 <absolute path to>Logic tensile-output HIP
+
+
+.. \-\-cmake-cxx-compiler=CMAKECXXCOMPILER
+    This doesn't appear to do much and I would like to consider removing
```
Comment (bstefanuk) [inline docs/src/cli-reference/TensileCreateLibrary.rst:133]: Should we delete this from the file, mark as deprecated, or keep it and move it back up to the **Options** section?

File: docs/src/cli-reference/TensileCreateLibrary.rst [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+Options
+-------
+
+When invoking *TensileCreateLibrary*, one can provide zero or more options.
+
+\-\-architecture=ARCHITECTURE
+    Architectures to generate a library for. When specifying multiple options, use quoted, semicolon delimited 
+    architectures, e.g., --architecture='gfx908;gfx1012'.
+    Supported architechures include: all gfx000 gfx803 gfx900 gfx900:xnack- gfx906 gfx906:xnack+ gfx906:xnack- gfx908 gfx908:xnack+
```
Comment (bstefanuk) [inline docs/src/cli-reference/TensileCreateLibrary.rst:37]: I think this is my typo but still a typo, should be "architectures"
```suggestion
    Supported architectures include: all gfx000 gfx803 gfx900 gfx900:xnack- gfx906 gfx906:xnack+ gfx906:xnack- gfx908 gfx908:xnack+
```

File: docs/src/cli-reference/TensileCreateLibrary.rst [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+\-\-ignore-asm-cap-cache
+    Ignore asm capability cache and derive the asm capabilities at runtime.    
+\-\-jobs=CPUTHREADS, \-j CPUTHREADS
+    Number of parallel jobs to launch. If this options is set higher than *nproc* the number of parallel 
+    jobs will be equal to the number of cores. If the this option is set below 1 (e.g. 0 or -1), the number
+    of parallel jobs will be set to the number of cores, up to a maximum of 64. (default = -1).    
+\-\-lazy-library-loading
+    Loads Tensile libraries when needed instead of upfront.
+\-\-library-format={yaml,msgpack}
+    Select which library format to use (default: msgpack).
```
Comment (bstefanuk) [inline docs/src/cli-reference/TensileCreateLibrary.rst:70]: For consistency, let's either change this to `(default = msgpack)` or change L66 above to `(default: -1)`, you can choose.

File: docs/src/cli-reference/TensileCreateLibrary.rst [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+\-\-library-print-debug
+    Solutions will print enqueue info when enqueueing a kernel.
```
Comment (bstefanuk) [inline docs/src/cli-reference/TensileCreateLibrary.rst:72]: Enqueue info? Is it worth expanding the meaning of this term?

File: docs/src/cli-reference/TensileCreateLibrary.rst [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+    Store every solution and kernel in separate file.
+\-\-no-short-file-names
+    Disables short files names.
+\-\-num-merged-files=NUMMERGEDFILES
+    Number of files the kernels should be written into.
+\-\-merge-files
+    Store all solutions in single file (default).
+\-\-package-library 
+    Enable the packaging of the client library code objects into separate (deprecated).
+    architectures as sub-directories within the Tensile/library directory.
```
Comment (bstefanuk) [inline docs/src/cli-reference/TensileCreateLibrary.rst:87]:
```suggestion
\-\-package-library 
    Enable the packaging of the client library code objects into separate
    architectures as sub-directories within the Tensile/library directory (deprecated).
```

File: docs/src/cli-reference/TensileCreateLibrary.rst [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+\-\-short-file-names
+    On windows kernel names can get too long.
```
Comment (bstefanuk) [inline docs/src/cli-reference/TensileCreateLibrary.rst:89]:
```suggestion
    On Windows, kernel names can get too long. 
```

File: docs/src/cli-reference/TensileCreateLibrary.rst [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+\-\-package-library 
+    Enable the packaging of the client library code objects into separate (deprecated).
+    architectures as sub-directories within the Tensile/library directory.
+\-\-short-file-names
+    On windows kernel names can get too long. 
+    Converts solution and kernel names to serial ids (default).
+\-\-separate-architectures
+    Separates TensileLibrary file by architecture to reduce the time to load the library file.
+    The separate-architectures option writes each architecture into a different TensileLibrary_gfxXXX.dat 
+    file.
```
Comment (bstefanuk) [inline docs/src/cli-reference/TensileCreateLibrary.rst:94]:
```suggestion
    Separates TensileLibrary file by architecture to reduce the time to load the library file.
    This option writes each architecture into a different TensileLibrary_gfxXXX.dat file.
```

File: docs/src/cli-reference/TensileCreateLibrary.rst [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+
+.. code-block::
+
+    TensileCreateLibrary <absolute path to>/Logic tensile-output HIP
+
+Adding TensileCreateLibrary options 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following example illustrates how to add options When
+invoking *TensileCreateLibrary*. In some cases, such as ``--seprate-architectures``
```
Comment (bstefanuk) [inline docs/src/cli-reference/TensileCreateLibrary.rst:124]:
```suggestion
invoking *TensileCreateLibrary*. In some cases, such as ``--separate-architectures``
```

File: docs/src/cli-reference/TensileCreateLibrary.rst [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+will contain the artifacts.
+
+.. code-block::
+
+    TensileCreateLibrary <absolute path to>/Logic tensile-output HIP
+
+Adding TensileCreateLibrary options 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following example illustrates how to add options When
```
Comment (bstefanuk) [inline docs/src/cli-reference/TensileCreateLibrary.rst:123]:
```suggestion
The following example illustrates how to add options when
```

File: docs/src/cli-reference/TensileCreateLibrary.rst (hunk 6 above)
Comment (ellosel) [reply docs/src/cli-reference/TensileCreateLibrary.rst:133]: Yes I would like to remove it.

File: docs/src/cli-reference/TensileCreateLibrary.rst (hunk 9 above)
Comment (ellosel) [reply docs/src/cli-reference/TensileCreateLibrary.rst:72]: After taking a longer look at this option, I am not sure that it does anything. We should consider removing.

File: docs/src/cli-reference.rst (hunk 5 above)
Comment (ellosel) [reply docs/src/cli-reference.rst:30]: {a, b, c} -> user must choose a or b or c. Suggestions?

File: docs/src/cli-reference/TensileCreateLibrary.rst (hunk 10 above)
Comment (ellosel) [reply docs/src/cli-reference/TensileCreateLibrary.rst:87]: Thoughts on noting that undocumented options are not considered supported?

Comment (bstefanuk) [reply docs/src/cli-reference/TensileCreateLibrary.rst]: I think it's a good idea, we should also then add `deprecated` to the help docs directly.

File: docs/src/cli-reference.rst (hunk 5 above)
Comment (bstefanuk) [reply docs/src/cli-reference.rst]: Makes sense, that sounds good to me

**Review by bstefanuk — APPROVED** [review]
Looks good, feel free to pick a solution for the unresolved comments. Approved.

**Review by babakpst — APPROVED** [review]
Thanks for the update. Please squash and merge.

**Pull request merged**